dnsapi/dns_oci.sh

```diff
@@ -0,0 +1,246 @@
+#!/usr/bin/env sh
+#
+# Acme.sh DNS API plugin for Oracle Cloud Infrastructure
+# Copyright (c) 2021, Oracle and/or its affiliates
+#
+# Required environment variables:
+# - OCI_TENANCY    : OCID of tenancy that contains the target DNS zone
+# - OCI_USER       : OCID of user with permission to add/remove records from zones
+# - OCI_FINGERPRINT: fingerprint of the public key for the user
+# - OCI_PRIVATE_KEY: Path to private API signing key file in PEM format
+#
+# Optional environment variables:
+# - OCI_KEY_PASSPHRASE: if the private key above s encrypted, the passphrase is required
+# - OCI_REGION: Your home region will probably response the fastest
+#
+
+dns_oci_add() {
+  _fqdn="$1"
```

Comment:
You can check if  `oci` is already installed on the machine,  if installed, you can prefer to use oci instead of curl.

And you can also another optional env variable:  `OCI_FORCE_GET` to force to use `curl`, default is empty,

Reply:
I specifically don't use `oci` so that it's completely portable to platforms on which `oci` has not been tested/ported. There's no real benefit to using `oci` now, as it's actually slightly slower than the method I'm using (because it has to load in all of the Python SDK).

Reply:
In case the `oci` is already installed and configured,  the user can just install acme.sh and use it, he doesn't need to configure for acme.sh again.

It's possible that the user will change the private key in futuren, he won't want to update twice.

And I believe `oci` is more safe to use.

Reply:
this is the approach we use in the dns aws plugin.

Reply:
Fair enough. I'll add `oci` support at the same time as removing the need to specify a fingerprint.

Reply:
I just want to double-check this because I've reviewed `dns_aws.sh` and as far as I can tell, it only uses `_get()` and `_post()`. I'm happy to add support for using the OCI CLI, I just wanted to be sure that's what you meant.